Assets/Scripts/Abilities.cs

```diff
@@ -214,4 +214,328 @@ private IEnumerator _flashlightAbility(PlayerCharacterStats stats)
             EnableButtons();
         }
     }
+
+
+	public void CoveringFire(PlayerCharacterStats stats) { if (stats.Actionsleft > 0) StartCoroutine(_CoveringFire(stats)); }
+	private IEnumerator _CoveringFire(PlayerCharacterStats stats)
+	{
+		try {
+			DisableButtons();
+			var abilData = (from abil in stats.AbilityData where abil.Name == "CoveringFire" select abil).FirstOrDefault();
+			//Debug.Log(abilData.Description);
+			GetComponent<PlayerMovementManager>().SetQuadsEnabled(false);
+			GetComponent<PlayerMovementManager>().enabled = false;
+			do
+			{
+				if (Input.GetKeyDown(KeyCode.Escape)) yield break;
+				yield return null;
+			} while (!Input.GetMouseButtonDown(0));
+			stats.isCoveringFire = true;
+			abilData.Currcooldown = abilData.Maxcooldown;
+			//stats.hasMoved = true;
+			stats.Actionsleft--;
+			GetComponent<TurnManager>().AutoEndTurnCheck();
```

Comment:
Why is this check being made?

Reply:
This is the same check as the other abilities making sure players don't move while trying to confirm the ability and to auto check for the end of the turn.  Not shown here but this also now does the UI for the ability.